modrinth/updater.go

```diff
@@ -0,0 +1,106 @@
+package modrinth
+
+import (
+	"errors"
+
+	"github.com/comp500/packwiz/core"
+	"github.com/mitchellh/mapstructure"
+)
+
+type mrUpdateData struct {
+	ModID            string `mapstructure:"mod-id"`
+	Versions         int    `mapstructure:"versions"`
+	InstalledVersion string `mapstructure:"installed-version"`
+}
+
+func (u mrUpdateData) ToMap() (map[string]interface{}, error) {
+	newMap := make(map[string]interface{})
+	err := mapstructure.Decode(u, &newMap)
+	return newMap, err
+}
+
+type mrUpdater struct{}
+
+func (u mrUpdater) ParseUpdate(updateUnparsed map[string]interface{}) (interface{}, error) {
+	var updateData mrUpdateData
+	err := mapstructure.Decode(updateUnparsed, &updateData)
+	return updateData, err
+}
+
+type cachedStateStore struct {
+	Mod     Mod
+	Version Version
+}
+
+func (u mrUpdater) CheckUpdate(mods []core.Mod, mcVersion string) ([]core.UpdateCheck, error) {
+	results := make([]core.UpdateCheck, len(mods))
+
+	pack, err := core.LoadPack()
+	if err != nil {
+		return results, err
+	}
+
+	for i, mod := range mods {
+		rawData, ok := mod.GetParsedUpdateData("modrinth")
+		if !ok {
+			results[i] = core.UpdateCheck{Error: errors.New("couldn't parse mod data")}
+			continue
+		}
+
+		data := rawData.(mrUpdateData)
+
+		fetchedMod, err := fetchMod(data.ModID)
+		if err != nil {
+			results[i] = core.UpdateCheck{Error: err}
+			continue
+		}
+
+		if len(fetchedMod.Versions) == data.Versions {
+			//the amount of listed versions hasn't changed. There shouldn't be any update
+			results[i] = core.UpdateCheck{UpdateAvailable: false}
+			continue
+		}
+
+		latestVersion, err := fetchedMod.fetchAndGetLatestVersion(pack)
```

Comment:
You could probably reuse the mcVersion given to you here (just a nitpick really, getting the version again isn't tooo bad)